### PR TITLE
docs: update kargo quickstart guide

### PR DESCRIPTION
**Pull request by fykaa**
https://docs.kargo.io/quickstart

### CODE REVIEW

File: docs/docs/20-quickstart.md [hunk 1]
```diff
@@ -432,7 +444,7 @@ the previous section.
     kargo-demo           13s
     ```
 
-1. Use the CLI to view our three `Stage` resources:
+5. Use the CLI to view our three `Stage` resources:
```
Comment (krancour) [inline docs/docs/20-quickstart.md:447]:
```suggestion
1. Use the CLI to view our three `Stage` resources:
```

File: docs/docs/20-quickstart.md [hunk 2]
```diff
@@ -475,14 +487,17 @@ the previous section.
     version of the `public.ecr.aws/nginx/nginx` container image.
     :::
 
-1. We'll use it later, so save the ID of the `Freight` to an environment
+7. We'll use it later, so save the ID of the `Freight` to an environment
```
Comment (krancour) [inline docs/docs/20-quickstart.md:490]:
```suggestion
1. We'll use it later, so save the ID of the `Freight` to an environment
```

File: docs/docs/20-quickstart.md [hunk 3]
```diff
@@ -419,7 +431,7 @@ the previous section.
   Kargo uses [semver](https://github.com/masterminds/semver#checking-version-constraints) to handle semantic versioning constraints.
   :::
 
-1. Use the CLI to view our `Warehouse` resource:
+4. Use the CLI to view our `Warehouse` resource:
```
[the remark this answers is not in the document]
Comment (krancour) [reply docs/docs/20-quickstart.md:434]: Ahhhhh...

But I see now that numbering seemed to reset here...

That must be due to a formatting issue somewhere above here that interrupted the ordered list.

We should find and fix the root issue instead.

File: docs/docs/20-quickstart.md [hunk 4]
```diff
@@ -475,14 +487,17 @@ the previous section.
...
 
     ```shell
     export FREIGHT_ALIAS=$(kargo get freight --project kargo-demo --output jsonpath={.alias})
     ```
 
-1. Now, let's _promote_ the `Freight` into the `test` `Stage`:
+     For additional details and a visual overview, please refer to the Kargo dashboard.
+  ![Argo-dashboard-screenshot](../static/img/kargo-dashboard-projects.png)
+
+8. Now, let's _promote_ the `Freight` into the `test` `Stage`:
```
Comment (krancour) [inline docs/docs/20-quickstart.md:500]:
```suggestion
1. Now, let's _promote_ the `Freight` into the `test` `Stage`:
```

File: docs/docs/20-quickstart.md [hunk 5]
```diff
@@ -643,7 +662,7 @@ the previous section.
     ]
     ```
 
-1. If we look at our `Freight` in greater detail, we'll see that by virtue of
+9. If we look at our `Freight` in greater detail, we'll see that by virtue of
```
Comment (krancour) [inline docs/docs/20-quickstart.md:665]:
```suggestion
1. If we look at our `Freight` in greater detail, we'll see that by virtue of
```

File: docs/docs/20-quickstart.md [hunk 6]
```diff
@@ -696,7 +715,7 @@ our `uat` and `prod` `Stage`s both subscribe to other, _upstream_ `Stage`s,
 thereby forming a _pipeline_:
 
 1. `uat` subscribes to `test`
-1. `prod` subscribes to `uat`.
+2. `prod` subscribes to `uat`.
```
Comment (krancour) [inline docs/docs/20-quickstart.md:718]:
```suggestion
1. `prod` subscribes to `uat`.
```

File: docs/docs/20-quickstart.md [hunk 7]
```diff
@@ -60,6 +60,12 @@ for Docker Desktop for Mac OS only. You can follow
 [these instructions](https://docs.orbstack.dev/kubernetes/) to enable its
 built-in Kubernetes support.
 
+To ensure that your Kubernetes cluster is up and running, use the following command:
+
+```shell
+orb start k8s
+```
+
```
Comment (krancour) [inline docs/docs/20-quickstart.md:68]: I am supportive of this kind of improvement, but let's spike it out into a separate PR. If we start being more detailed about how to start it, we probably need to be more detailed about stopping/deleting it, with applicable disclaimers, etc. It just begins to seem like its own discrete issue.

File: docs/docs/20-quickstart.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (krancour) [reply docs/docs/20-quickstart.md:434]: It's got to be something that is not indented properly.

**Comment by krancour** [comment]
General comment -- this is a modest improvement. 👍 

Let's also plan, however, on providing more UI-centric instructions. i.e. Instead of just "this is what it will look like when you click this link," we also need "click this, then click that," as an alternative to the CLI-based instructions.

File: docs/docs/20-quickstart.md [hunk 8]
```diff
@@ -415,9 +421,9 @@ the previous section.
     EOF
     ```
     
-  :::info
-  Kargo uses [semver](https://github.com/masterminds/semver#checking-version-constraints) to handle semantic versioning constraints.
-  :::
+      :::info
+      Kargo uses [semver](https://github.com/masterminds/semver#checking-version-constraints) to handle semantic versioning constraints.
+      :::
```
Comment (krancour) [inline docs/docs/20-quickstart.md:426]: Ahh. This info box is what broke the numbering, huh? 👍

**Comment by krancour** [comment]
> Regarding the need for more clarity on clicks, I believe that adding multiple screenshots inline may not be the most user-friendly approach.

I agree. So let's not constrain the effort of providing the quickstart with a UI perspective to just adding screenshots. There are other effective ways to document navigation of a UI.

This page has an example of how (stylystically speaking) we have documented certain steps on GitHub:

https://kargo.akuity.io/how-to-guides/managing-credentials/#github-app-authentication

There is also information about how to do this in the "Referencing UI elements" section of the style guide:

https://github.com/akuity/kargo/blob/main/docs/STYLE_GUIDE.md#referencing-ui-elements


> Should I consider using Supademo to illustrate each component instead?

I want to see how far we can get without falling back on this. I'm entirely unopposed to it as a _supplement_ to proper docs, but I don't want to skimp on the text-based docs specifically because they're ingestible by search engines and LLMs.

File: docs/docs/20-quickstart.md [hunk 9]
```diff
@@ -146,11 +146,15 @@ At the end of this process:
 * The Argo CD dashboard will be accessible at [localhost:31443](https://localhost:31443).
 
   The username and password are both `admin`.
+  
+  ![Argo-dashboard-screenshot](../static/img/argo-dashboard-1.png)
```
Comment (krancour) [inline docs/docs/20-quickstart.md:150]: I don't feel like this screenshot is contributing much. It's basically a blank page, so I'm not sure what purpose it serves other than to back up the claim that something is indeed running as localhost:31443.

I would consider dropping this one.

File: docs/docs/20-quickstart.md [hunk 10]
```diff
@@ -146,11 +146,15 @@ At the end of this process:
...
 
 * The Kargo dashboard will be accessible at [localhost:31444](https://localhost:31444).
 
   The admin password is `admin`.
 
+  ![Kargo-dashboard-screenshot](../static/img/kargo-dashboard-1.png)
```
Comment (krancour) [inline docs/docs/20-quickstart.md:156]: Same as [this comment](https://github.com/akuity/kargo/pull/2521/files#r1791978562)

File: docs/docs/20-quickstart.md [hunk 11]
```diff
@@ -261,6 +265,8 @@ all three Argo CD `Application`s have not yet synced because they're not
 configured to do so automatically, and in fact, the branches referenced by their
 `targetRevision` fields do not even exist yet.
 
+![Argo-dashboard-screenshot](../static/img/argo-dashboard-2.png)
```
Comment (krancour) [inline docs/docs/20-quickstart.md:268]: Compared to the prior two screenshots, I am more ok with this one because its more than just a blank page and gives the readers a quick and easy way to assert the success of previous steps.

File: docs/docs/20-quickstart.md [hunk 12]
```diff
@@ -482,6 +488,18 @@ the previous section.
...
+
+   1. Navigate to the Kargo Dashboard by clicking the link above or entering the URL directly in your browser.
+   1. In the left-hand menu, select the <Hlt>Projects</Hlt> section.
+   1. Choose your `Project`, <Hlt>kargo-demo</Hlt>, from the list of available `Project`s.
+   1. Once there, you will see a detailed visual representation of the `Project`. This includes:
+      1. The <Hlt>Warehouse</Hlt> and its status.
+      1. Various `Stage`s: <Hlt>test</Hlt>, <Hlt>uat</Hlt>, and <Hlt>prod</Hlt>.
+
+      ![Argo-dashboard-screenshot](../static/img/kargo-dashboard-projects.png)
+
```
Comment (krancour) [inline docs/docs/20-quickstart.md:502]: You have this positioned in a section titled "Hands on with the Kargo CLI," which makes this feel somewhat misplaced, but we can fix that...

What I suggest is breaking up the whole "Hands on with the Kargo CLI" section into two separate sections for "Define Your First Pipeline" and "Promote to Test."

"Define Your First Pipeline" should probably remain mostly (but not entirely) CLI-centric for the simple reason that it involves a lot of YAML and environment variables and UI-based instructions are going to make this task harder rather than easier. This tracks with expected/observed usage. Since pipelines are defined declaratively, people tend to create them with `kargo apply`, `kubectl apply`, or via gitops. I think you could even introduce the section by _saying_ that the CLI is the most convenient way to define the pipeline. That said, I believe that once you get past `kargo apply`, you can probably swap over to UI-based instructions. Instead of inviting users to `kargo get freight` or `kargo get stages`, you can walk them through using the UI to see the pipeline they just created, witness the first piece of Freight popping up in the Freight timeline, etc.

"Promote to Test" being pretty much exclusively about imperative actions (and not declarative configuration), you could just cut over, completely to UI-based instructions.

Anticipating a question about whether this means few CLI examples in the docs overall -- in the short term, it does. But I believe detailed usage of CLI commands will reappear over time in individual guides. Let's keep the focus of the quickstart on imparting a maximum amount of knowledge with minimal reader effort.

**Comment by krancour** [comment]
Made a few comments. I don't want to _just_ add screenshots. I think for this to make sense, we need a comprehensive restructuring of a few sections.

File: docs/docs/20-quickstart.md [hunk 13]
```diff
@@ -508,52 +510,28 @@ the previous section.
...
-
-    ```shell
-    kargo get warehouses --project kargo-demo
-    ```
+      :::info
+      Kargo uses [semver](https://github.com/masterminds/semver#checking-version-constraints) to handle semantic versioning constraints.
+      :::
 
-    Sample output:
+1. After applying the above configuration, you can view the pipeline you just created using the [Kargo Dashboard](https://localhost:31444/).
```
Comment (krancour) [inline docs/docs/20-quickstart.md:517]: Since this is an ordered list of steps that the reader is expected to follow, I think it's well-understood that this step comes after the previous one and we don't need to waste words on that.

```suggestion
1. View the pipeline you just created using the [Kargo Dashboard](https://localhost:31444/):
```

File: docs/docs/20-quickstart.md [hunk 14]
```diff
@@ -508,52 +510,28 @@ the previous section.
...
-    ```shell
-    NAME         SHARD   AGE
-    kargo-demo           13s
-    ```
+   1. Choose your `Project`, <Hlt>kargo-demo</Hlt>, from the list of available `Project`s
+   1. You can find the pipeline you've created in the center of your Project screen.
+   1. The pipeline consists of connected rectangles representing:
+      1. <Hlt>Subscription</Hlt>: Has information about the container image repository that the `Warehouse` is subscribed to.
+      1. <Hlt>kargo-kemo</Hlt>: This represents your `Warehouse`. It will automatically subscribe to the image and should produce `Freight` shortly.
+      1. <Hlt>test</Hlt>, <Hlt>uat</Hlt>, and <Hlt>prod</Hlt>: These are the deployment `Stage` resources.
```
Comment (krancour) [inline docs/docs/20-quickstart.md:524]: If this is someone's first time through the quickstart (and I think that's a safe thing to assume), they probably need to log into the dashboard.

This section also reads oddly, as it starts with imperatives ("do this") and then jumps immediately into information the user _doesn't need to act on_, but they're not really presented any differently than the things they do. You don't want someone wondering, "Am I supposed to do something here?"

```suggestion
   1. Log in using the password `admin`.
   
       This will take you to the home page of the dashboard.
       
       <Good place for a screen shot>
   
   1. Select your Project, <Hlt>kargo-demo</Hlt>, from the list.
   
       Here you can see a detailed overview of the Project we created using the CLI, including:
       
           * An interactive, visual representation of your pipeline:
               * A container image repository
               * A Warehouse that discovers new images as they are pushed to the repository.
               * Three Stages representing distinct instances of our demo application

           <Good place for a screen shot. Maybe highlight the pipeline.>

           * And interactive Freight timeline:
               * This displays collections of artifacts that have been discovered by the Warehouse.
               * Collections (Freight) are ordered chronologically, with newer Freight on the left and older Freight on the right.
                     
           <Good place for a screen shot. Maybe highlight the Freight timeline.>
```

The above suggestion may still need some formatting fixes and such.

File: docs/docs/20-quickstart.md [hunk 15]
```diff
@@ -261,13 +261,15 @@ all three Argo CD `Application`s have not yet synced because they're not
...
+![Argo-dashboard-screenshot](../static/img/argo-dashboard.png)
+
 :::info
 Our three stages all existing in a single cluster is for the sake of expediency.
 A single Argo CD control plane can manage multiple clusters, so these could just
 as easily have been spread across multiple clusters.
 :::
 
-### Hands on with the Kargo CLI
+### Define Your First Pipeline
```
Comment (krancour) [inline docs/docs/20-quickstart.md:272]: There's enough in both the YAML and explanatory text that follows that it might be better to make this heading slightly less specific:

```suggestion
### Your First Kargo Project
```

File: docs/docs/20-quickstart.md [hunk 16]
```diff
@@ -508,52 +510,28 @@ the previous section.
...
-  :::
-
-1. Use the CLI to view our `Warehouse` resource:
-
-    ```shell
-    kargo get warehouses --project kargo-demo
-    ```
+      :::info
+      Kargo uses [semver](https://github.com/masterminds/semver#checking-version-constraints) to handle semantic versioning constraints.
+      :::
```
Comment (krancour) [inline docs/docs/20-quickstart.md:515]: Let's ditch this. I put this here, but it feels out of place now.

We can move this to be inline in the Go/CRD docs. (You can make a separate issue for that.)

File: docs/docs/20-quickstart.md [hunk 17]
```diff
@@ -508,52 +510,28 @@ the previous section.
...
-    test           0/1 Fulfilled              NotApplicable   19s
-    uat            0/1 Fulfilled              NotApplicable   19s
-    ```
+   ![Kargo-dashboard-screenshot](../static/img/kargo-dashboard-projects.png)
 
 1. After a few seconds, our `Warehouse`, which subscribes to the
    `public.ecr.aws/nginx/nginx` container image, also should have already
-   produced `Freight`:
+   produced `Freight`. You can monitor the status of `Freight` in the various stages directly on the interface, 
+   where it will appear under the respective stages (`test`, `uat`, `prod`).
```
Comment (krancour) [inline docs/docs/20-quickstart.md:531]: I would reword this to say that after a few seconds, a piece of `Freight` should appear in the `Freight` timeline, if it isn't there already. Note that this _might_ not refresh automatically and might require a note mentioning the need for a hard refresh. Annoying as this could prove, such a note should also mention that the issue will soon be fixed.

cc @Marvin9

File: docs/docs/20-quickstart.md [hunk 18]
```diff
@@ -508,52 +510,28 @@ the previous section.
...
 
-    ```shell
-    kargo get freight --project kargo-demo
-    ```
-
-    Sample output:
+   The top of the screen displays a Freight Timeline. As your application progresses through the pipeline, the timeline will update to show you the current `Stage` of your `Freight`.
```
Comment (krancour) [inline docs/docs/20-quickstart.md:533]: > The top of the screen displays a Freight Timeline.

I think this has been established and can be removed.

> As your application progresses through the pipeline, the timeline will update to show you the current `Stage` of your `Freight`.

I would save this bit of information for _after_ the first Promotion and reword it to call attention to Freight in the Timeline being color-coded to reflect what Stages are currently using it.

File: docs/docs/20-quickstart.md [hunk 19]
```diff
@@ -568,269 +546,42 @@ the previous section.
...
-
-    Sample output:
-
-    ```shell
-    promotion.kargo.akuity.io/test.01j2wbtrym4r3tktv38qzteh5h.7a6e91f promotion created
-    ```
-
-    Query for `Promotion` resources within our project to see one has been
-    created:
+### Promote to Test
```
Comment (krancour) [inline docs/docs/20-quickstart.md:549]: To match my suggestion earlier of "Your First Project" we might consider renaming this to match:

```suggestion
### Your First Promotion
```

File: docs/docs/20-quickstart.md [hunk 20]
```diff
@@ -568,269 +546,42 @@ the previous section.
...
 
-    ```shell
-    kargo get promotions --project kargo-demo
-    ```
+1. To promote `Freight` to the `test` `Stage`, select the existing `Freight` from
+    the Freight Timeline above. Click the three dots, choose <Hlt>Promote</Hlt>,
+    and then select <Hlt>Promote</Hlt> that appeared on the `test` `Stage`.
```
Comment (krancour) [inline docs/docs/20-quickstart.md:553]: This is one of two ways to do it. You can pick a piece of Freight and then the Stage to deploy it to (as you are doing here) or you can pick a Stage and then the Freight you wish to deploy to it.

The latter is more "classic" behavior, with the former having been introduced more recently. To some users, one approach will feel more natural, to others, the other approach.

imho, the "stage first" approach will probably be more intuitively understood by a higher proportion of readers than the "freight first" approach.

File: docs/docs/20-quickstart.md [hunk 21]
```diff
@@ -568,269 +546,42 @@ the previous section.
...
+ 
+    The promotion may briefly appear in a `Pending` state, but it will likely
+    transition to `Running`, or even `Succeeded` within moments, as shown below:
```
Comment (krancour) [inline docs/docs/20-quickstart.md:556]: This wording probably made more sense with the CLI where you might need to issue a command to check on the status of the promotion a few times. The UI auto-refreshes and this is now more of a "blink and you may miss it" situation.

I would reword this, perhaps by taking some details you currently have farther down and moving them up here. Call out that when complete, there will be a checkmark. Call out that _after_ the Promotion is complete, a healthcheck will run periodically and when the Stage reaches a healthy state, it will be reflected with a heart.

File: docs/docs/20-quickstart.md [hunk 22]
```diff
@@ -568,269 +546,42 @@ the previous section.
...
 
-    Our `Promotion` may briefly appear to be in a `Pending` phase, but more than
-    likely, it will almost immediately be `Running`, or even `Succeeded`:
+    ![Kargo-dashboard-screenshot](../static/img/kargo-dashboard-promotion.png)
 
-    ```shell
-    NAME                                      SHARD   STAGE   FREIGHT                                    PHASE       AGE
-    test.01j92m3mx7jn49rc5p62tmnsf1.b2ceb4f           test    b2ceb4f821be8b682861c579171874028e4275fd   Succeeded   57s
-    ```
+    You'll notice a **_check mark_**, indicating a <Hlt>successful promotion</Hlt>.
```
Comment (krancour) [inline docs/docs/20-quickstart.md:560]: There may be several spots to fix this throughough, but please use `<Hlt>` only in _direct reference_ to some UI element. (There is no text anywhere on the screen that says "successful promotion".)

**Comment by krancour** [comment]
Additional notes that I couldn't add inline because they weren't near any changes...

If we're making this more UI-centric, I think we need some _brief_ explanatory text around line 289 (just before logging in using the Kargo CLI) about why we're using the CLI: There's a lot of configuration and it's easiest to create it declaratively.

I also think that section might be a good place to add some tabs showing how that step can be accomplished with either the Kargo CLI or kubectl. (kubectl is actually easier because there's no logging in required.) If you do this, you could also edit the section on installing the CLI so that it's designated as an _optional_ step.

File: docs/docs/20-quickstart.md [hunk 23]
```diff
@@ -507,53 +509,37 @@ the previous section.
...
-   `public.ecr.aws/nginx/nginx` container image, also should have already
-   produced `Freight`:
-
-    ```shell
-    kargo get freight --project kargo-demo
-    ```
+    :::note
+    Please note that the timeline _may_ not refresh automatically, and you might need to perform a hard refresh to see the updates. 
+    
+    We acknowledge that this can be inconvenient, but rest assured that this issue will be addressed in a future update.
```
Comment (krancour) [inline docs/docs/20-quickstart.md:539]: This somehow feels _overly_ conciliatory.

```suggestion
    Note that the timeline _may_ not refresh automatically and you may need to refresh the page to see new `Freight`.
    
    This inconvenience will be addressed in a forthcoming update.
```

File: docs/docs/20-quickstart.md (hunk 17 above)
Comment (Marvin9) [reply docs/docs/20-quickstart.md]: Sorry I didn't get the issue. Can you elaborate how the UI does not represent the accurate availability of freights as in controller?
